src/utils/spawner.js

```diff
@@ -41,71 +36,67 @@ var spawner = {
+            spawnUpgrader();
         }
 
-        if(builders.length < targetBuilders) {
-            var bodyPieces = [MOVE, MOVE, WORK];
-            var cost = body.MOVE.cost * 2 + body.WORK.cost;
-            while(cost + body.CARRY.cost <= availableEnergy){
-                bodyPieces.push(CARRY);
-                cost += body.CARRY.cost;
-            }
-            var code = spawner.spawnCreep(bodyPieces, createName(roles.BUILD), {memory: {role: roles.BUILD}});
-            console.log('Spawning new builder: ' + code);
-            return;
+        else if(builders.length < targetBuilders) {
+            console.log(`Harv: ${harvesters.length}/${targetHarvesters} Haul: ${haulers.length}/${targetHaulers} Buil: ${builders.length}/${targetBuilders} Upgr: ${upgraders.length}/${targetUpgraders}`);
+            spawnBuilder();
         }
 
-        if(haulers.length < targetHaulers) {
-            var bodyPieces = [MOVE, CARRY];
-            var cost = body.MOVE.cost + body.CARRY.cost;
-            while(cost + body.WORK.cost + body.CARRY.cost <= availableEnergy){
-                bodyPieces.push(WORK);
-                bodyPieces.push(CARRY);
-                cost += body.WORK.cost + body.CARRY.cost;
-            }
-            var code = spawner.spawnCreep(bodyPieces, createName(roles.HAUL), {memory: {role: roles.HAUL}});
-            console.log('Spawning new hauler: ' + code);
-            return;
+        else if(haulers.length < targetHaulers) {
+            console.log(`Harv: ${harvesters.length}/${targetHarvesters} Haul: ${haulers.length}/${targetHaulers} Buil: ${builders.length}/${targetBuilders} Upgr: ${upgraders.length}/${targetUpgraders}`);
+            spawnHauler();
+        }
+
+        else {
+            //targetHarvesters += 1;
+            targetBuilders = 2;
+            targetUpgraders = 2;
+            targetHaulers = 2;
+        }
+
+        if(harvesters.length < targetHarvesters) {
+            console.log(`Harv: ${harvesters.length}/${targetHarvesters} Haul: ${haulers.length}/${targetHaulers} Buil: ${builders.length}/${targetBuilders} Upgr: ${upgraders.length}/${targetUpgraders}`);
+            spawnHarvester();
+        }
+        
+        else if(upgraders.length < targetUpgraders) {
+            console.log(`Harv: ${harvesters.length}/${targetHarvesters} Haul: ${haulers.length}/${targetHaulers} Buil: ${builders.length}/${targetBuilders} Upgr: ${upgraders.length}/${targetUpgraders}`);
+            spawnUpgrader();
+        }
+
+        else if(builders.length < targetBuilders) {
+            console.log(`Harv: ${harvesters.length}/${targetHarvesters} Haul: ${haulers.length}/${targetHaulers} Buil: ${builders.length}/${targetBuilders} Upgr: ${upgraders.length}/${targetUpgraders}`);
+            spawnBuilder();
         }
+
+        else if(haulers.length < targetHaulers) {
```

Comment:
We should have a think about making this long list of elseif's some generic config that can be passed in and parsed rather than hardcoded. I'll raise an issue for this, this'll do for now.

Reply:
issues/10